cabal-testsuite/PackageTests/Freeze/T9799b/src/None.hs

```diff
@@ -0,0 +1,10 @@
+{-# LANGUAGE TemplateHaskell #-}
+module None where
+
+import MyLib
+import Language.Haskell.TH
+
+$(do
+    runIO $ putStrLn $ "Building: " ++ renamedVers
+    [d| x = () |]
```

Comment:
`[d| x = () |]` will create a top-level declaration of `x = ()`. The key point here is that before doing so it will run code at compile time (the `runIO` part). This module builds because we'll qualify in the freeze file: `any.libA == 0.1.0.0 | == 0.2.0.0`, so we'll pick version 0.2.0.0 for building this module and for building setup. In this case we'll find `renamedVers`, but in the Setup case we'll fail to find `vers`.

Reply:
Thanks!